### PR TITLE
Fix SpectralModel.evaluate_error gradient computation

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request addresses #2601. I've change to the alternative method proposed in the [prototype notebook](https://github.com/gammapy/gammapy-extra/blob/master/experiments/uncertainty_estimation_prototype.ipynb), which allowed to use a separate eps per parameter.

### CODE REVIEW

File: gammapy/modeling/models/spectral.py [hunk 1]
```diff
@@ -47,34 +46,45 @@ def __sub__(self, model):
...
+
+    def evaluate_error(self, energy, precision=1e-4):
         """Evaluate spectral model with error propagation.
 
         Parameters
         ----------
         energy : `~astropy.units.Quantity`
             Energy at which to evaluate
+        precision : float
+            Numerical precision of the gradient evaluation.
```
Comment (cdeil) [inline gammapy/modeling/models/spectral.py]: Suggest to rename the parameter and description. It's not the precision of the gradient. It's the step size for the parameter.

File: gammapy/modeling/models/tests/test_spectral.py [hunk 2]
```diff
@@ -589,8 +589,21 @@ def test_ecpl_model(self):
...
         assert_allclose(out.data, [1.548176e-10, 1.933612e-11], rtol=1e-3)
 
-    def test_naima_model_error_proprgation(self):
-        # Regression test for Naima model
-        # https://github.com/gammapy/gammapy/issues/2190
-        # TODO: implement test case. Move to Naima model tests!
-        pass
+
+@requires_dependency("naima")
+def test_naima_model_error_propagation():
```
Comment (cdeil) [inline gammapy/modeling/models/tests/test_spectral.py]: Suggest to move the `evaluate_error` call the the Naima example we already have, as here:
https://github.com/gammapy/gammapy/pull/2602/files
Advantage: less test code.

File: gammapy/modeling/models/spectral.py (hunk 1 above)
Comment (adonath) [reply gammapy/modeling/models/spectral.py]: I renamed the parameter to `epsilon` (that's what we have for `SpectralModel.spectral_index()` as well) and changed the description accordingly.

File: gammapy/modeling/models/tests/test_spectral.py (hunk 2 above)
Comment (adonath) [reply gammapy/modeling/models/tests/test_spectral.py]: Done

**Pull request merged**